river/View.zig

```diff
@@ -117,6 +117,10 @@ saved_buffers: std.ArrayList(SavedBuffer),
 /// view returns to floating mode.
 float_box: Box = undefined,
 
+/// Pre-fullscreen dimensions of the view, saved so that they can be restored if
+/// the view returns from fullscreen without needing to rearrange the output.
+post_fullscreen_box: Box = undefined,
```

Comment:
```suggestion
/// While a view is in fullscreen, it is still arranged if a layout is active but the resulting dimensions are stored here instead of being applied to the view's state. This allows us to avoid an arrange when the view returns from fullscreen and for more intuitive behavior if there is no active layout for the output.
post_fullscreen_box: Box = undefined,
```